universal-application-tool-0.0.1/app/views/AzureFileUploadViewStrategy.java

```diff
@@ -0,0 +1,220 @@
+    }
+    BlobStorageUploadRequest request = (BlobStorageUploadRequest) storageUploadRequest;
+    ContainerTag formTag = form().withId("azure-upload-form-component");
+    formTag.with(
+        footer(viewUtils.makeWebJarsTag(AZURE_STORAGE_BLOB_WEB_JAR)),
+        footer(viewUtils.makeLocalJsTag("azure_upload")));
+    return formTag
+        .with(input().withType("file").withName("file"))
+        .with(input().withType("hidden").withName("sasToken").withValue(request.sasToken()))
+        .with(input().withType("hidden").withName("blobUrl").withValue(request.blobUrl()))
+        .with(
+            input().withType("hidden").withName("containerName").withValue(request.containerName()))
+        .with(input().withType("hidden").withName("fileName").withValue(request.fileName()))
+        .with(input().withType("hidden").withName("accountName").withValue(request.accountName()))
+        .with(
+            input()
+                .withType("hidden")
+                .withName("successActionRedirect")
+                .withValue(request.successActionRedirect()))
+        .with(TagCreator.button(text("Upload to Azure Blob Storage")).withType("submit"));
+  }
+
+  @Override
+  public Tag renderFileUploadBlockSubmitForms(
+      Params params, ApplicantQuestionRendererFactory applicantQuestionRendererFactory) {
+    String key = FileNameFormatter.formatFileUploadQuestionFilename(params);
+    String onSuccessRedirectUrl =
+        params.baseUrl()
+            + routes.ApplicantProgramBlocksController.updateFile(
+                    params.applicantId(),
+                    params.programId(),
+                    params.block().getId(),
+                    params.inReview())
+                .url();
+
+    StorageUploadRequest request =
+        params.storageClient().getSignedUploadRequest(key, onSuccessRedirectUrl);
+
+    if (!(request instanceof BlobStorageUploadRequest)) {
+      throw new RuntimeException(
+          "Tried to upload a file to Azure Blob storage using incorrect request type");
+    }
+    BlobStorageUploadRequest blobStorageUploadRequest = (BlobStorageUploadRequest) request;
+    ApplicantQuestionRendererParams rendererParams =
+        ApplicantQuestionRendererParams.builder()
+            .setMessages(params.messages())
+            .setSignedFileUploadRequest(blobStorageUploadRequest)
+            .build();
+
+    ContainerTag formTag =
+        form()
+            .withId(BLOCK_FORM_ID)
+            .with(
+                each(
+                    params.block().getQuestions(),
+                    question ->
+                        renderQuestion(
+                            question, rendererParams, applicantQuestionRendererFactory)));
+
+    Tag skipForms = renderDeleteAndContinueFileUploadForms(params);
+    Tag buttons = renderFileUploadBottomNavButtons(params);
+
+    return div(formTag, skipForms, buttons);
+  }
+
+  Tag renderFileUploadBottomNavButtons(Params params) {
```

Comment:
question (blocking): can this be private?

Reply:
Done.